src/axom/primal/geometry/Segment.hpp

```diff
@@ -39,20 +57,26 @@ template <typename T, int NDIMS>
 class Segment
 {
 public:
-  typedef Point<T, NDIMS> PointType;
+  using PointType = Point<T, NDIMS>;
+  using VectorType = Vector<T, NDIMS>;
+
+  enum
+  {
+    NUM_SEG_VERTS = 2
+  };
 
 public:
   /*!
    * \brief Creates a segment instance from point A to point B.
    * \param A user-supplied source point
    * \param B user-supplied target point
    */
-  Segment(const PointType& A, const PointType& B);
+  Segment(const PointType& A, const PointType& B) : m_source(A), m_target(B) {};
 
   /*!
    * \brief Destructor.
    */
-  ~Segment();
+  ~Segment() = default;
```

Comment:
I'd probably just delete this, if the intent is to make `Segment` a POD-ish type.